src/sidebar/util/test/groups-test.js

```diff
@@ -62,16 +78,141 @@ describe('sidebar.util.groups', () => {
     });
 
     it('handles case where there is no Public group', () => {
       const userGroups = [];
       const featuredGroups = [];
 
-      const groups = combineGroups(userGroups, featuredGroups);
+      const groups = combineGroups(
+        userGroups,
+        featuredGroups,
+        'https://foo.com/bar'
+      );
       assert.deepEqual(groups, []);
     });
+
+    it('adds isSelectable property to groups', () => {
+      const userGroups = [{ id: 'one', name: 'GroupA' }];
+      const featuredGroups = [{ id: '__world__', name: 'Public' }];
+
+      const groups = combineGroups(
+        userGroups,
+        featuredGroups,
+        'https://foo.com/bar'
+      );
+
+      groups.forEach(g => assert.equal(g.isSelectable, true));
+    });
+
+    it('sets isSelectable to true if scoped is falsey', () => {
+      const userGroups = [{ id: 'one', name: 'GroupA', scoped: false }];
+      const featuredGroups = [{ id: '__world__', name: 'Public' }];
+
+      const groups = combineGroups(
+        userGroups,
+        featuredGroups,
+        'https://foo.com/bar'
+      );
+
+      groups.forEach(g => assert.equal(g.isSelectable, true));
+    });
+
+    it('sets isSelectable to true if scopes.enforced is falsey', () => {
+      const userGroups = [
+        {
+          id: 'one',
+          name: 'GroupA',
+          scoped: true,
+          scopes: { enforced: false },
+        },
+      ];
+      const featuredGroups = [
+        { id: 'two', name: 'GroupB', scoped: true, scopes: {} },
+      ];
+
+      const groups = combineGroups(
+        userGroups,
+        featuredGroups,
+        'https://foo.com/bar'
+      );
+
+      groups.forEach(g => assert.equal(g.isSelectable, true));
+    });
+
+    it('sets isSelectable to true if scopes.uri_patterns match the uri', () => {
+      const userGroups = [
+        {
+          id: 'one',
+          name: 'GroupA',
+          scoped: true,
+          scopes: {
+            enforced: true,
+            uri_patterns: ['http://foo.com*', 'https://foo.com*'],
+          },
+        },
+      ];
+      const featuredGroups = [];
+
+      const groups = combineGroups(
+        userGroups,
+        featuredGroups,
+        'https://foo.com/bar'
+      );
+
+      groups.forEach(g => assert.equal(g.isSelectable, true));
+    });
+
+    it('sets isSelectable to false if scopes.uri_patterns do not match the uri', () => {
+      const userGroups = [
+        {
+          id: 'one',
+          name: 'GroupA',
+          scoped: true,
+          scopes: {
+            enforced: true,
+            uri_patterns: ['http://foo.com*'],
+          },
+        },
+      ];
+      const featuredGroups = [];
+
+      const groups = combineGroups(
+        userGroups,
+        featuredGroups,
+        'https://foo.com/bar'
+      );
+
+      groups.forEach(g => assert.equal(g.isSelectable, false));
+    });
+
+    it('it permits multiple *s in the scopes uri pattern', () => {
```

Comment:
Can you also add test(s) to verify that non-* chars in the pattern are escaped correctly?